nixos/modules/services/web-apps/immich-public-proxy.nix

```diff
@@ -0,0 +1,108 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.immich-public-proxy;
+  inherit (lib)
+    types
+    mkIf
+    mkOption
+    mkEnableOption
+    ;
+in
+{
+  options.services.immich-public-proxy = {
+    enable = mkEnableOption "Immich Public Proxy";
+    package = lib.mkPackageOption pkgs "immich-public-proxy" { };
+
+    immich-url = mkOption {
+      type = types.str;
+      description = "URL of the Immich instance";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 3000;
+      description = "The port that IPP will listen on.";
+    };
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Whether to open the IPP port in the firewall";
+    };
+    user = mkOption {
+      type = types.str;
+      default = "ipp";
+      description = "The user IPP should run as.";
+    };
+    group = mkOption {
+      type = types.str;
+      default = "ipp";
+      description = "The group IPP should run as.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.immich-public-proxy = {
+      description = "Immich public proxy for sharing albums publicly without exposing your Immich instance";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      #path = [ pkgs.immich-public-proxy ];
+      environment = {
+        IMMICH_URL = cfg.immich-url;
+        IPP_PORT = builtins.toString cfg.port;
+      };
+      serviceConfig = {
+        ExecStart = lib.getExe cfg.package;
+        StateDirectory = "ipp"; # TODO wat do
+        SyslogIdentifier = "ipp";
+        RuntimeDirectory = "ipp"; # TODO wat do
```

Comment:
Does IPP store anything in `/var/lib/ipp`?

Reply:
No, I copied that from the `immich` module but forgot to go back and figure out what it actually did. I'll remove those directory configs.